src/Middleware/Diagnostics/test/UnitTests/ExceptionHandlerTest.cs

```diff
@@ -149,7 +150,7 @@ public async Task ClearsResponseBuffer_BeforeRequestIsReexecuted()
                     app.Run(async (context) =>
                     {
                         // Write some content into the response before throwing exception
-                        await context.Response.WriteAsync(new string('a', 100));
+                        await context.Response.Body.WriteAsync(Encoding.UTF8.GetBytes(new string('a', 100)));
```

Comment:
Why did this change? These two lines should behave be identically.

Reply:
Response.WriteAsync flushes. Exception handler can't run after flush.

Reply:
I see, this is an odd test. The buffering prevented OnStarting from firing before, but now it fires directly from the WriteAsync extension. For this kind of buffering to really be effective it will need to suppress/wrapp the new StartAsync feature.

Reply:
What do you want to do? Can this be merged as is or does it require more changes?

Reply:
Revert this line and then go to line 120 and remove IHttpResponseStartFeature from the feature collection. That's what you'd have to do to make this buffering-to-prevent-start scenario work.  Then :shipit: